common/dynamicconfig/file_based_client.go

```diff
@@ -95,15 +95,16 @@ func NewFileBasedClient(config *FileBasedClientConfig, logger log.Logger, doneCh
 	return client, nil
 }
 
-func (fc *fileBasedClient) GetValue(name Key, defaultValue interface{}) (interface{}, error) {
-	return fc.getValueWithFilters(name, nil, defaultValue)
+func (fc *fileBasedClient) GetValue(name Key) (interface{}, error) {
+	return fc.getValueWithFilters(name, nil, name.DefaultValue())
 }
 
-func (fc *fileBasedClient) GetValueWithFilters(name Key, filters map[Filter]interface{}, defaultValue interface{}) (interface{}, error) {
-	return fc.getValueWithFilters(name, filters, defaultValue)
+func (fc *fileBasedClient) GetValueWithFilters(name Key, filters map[Filter]interface{}) (interface{}, error) {
+	return fc.getValueWithFilters(name, filters, name.DefaultValue())
 }
 
-func (fc *fileBasedClient) GetIntValue(name Key, filters map[Filter]interface{}, defaultValue int) (int, error) {
+func (fc *fileBasedClient) GetIntValue(name IntKey, filters map[Filter]interface{}) (int, error) {
+	defaultValue := name.DefaultInt()
```

Comment:
same DefaultValue() note

Reply:
we need to return int instead of interface{} in this function, cannot use DefaultValue() here

Reply:
what's the benefit here?